spring-integration-mqtt/src/test/java/org/springframework/integration/mqtt/Mqttv5AdapterTests.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.mqtt;
+
+
+import org.eclipse.paho.mqttv5.client.IMqttAsyncClient;
+import org.eclipse.paho.mqttv5.client.IMqttMessageListener;
+import org.eclipse.paho.mqttv5.client.IMqttToken;
+import org.eclipse.paho.mqttv5.client.MqttConnectionOptions;
+import org.eclipse.paho.mqttv5.common.MqttException;
+import org.eclipse.paho.mqttv5.common.MqttSubscription;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.integration.channel.NullChannel;
+import org.springframework.integration.mqtt.inbound.Mqttv5PahoMessageDrivenChannelAdapter;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+/**
+ * @author Gary Russell
+ * @author Artem Bilan
+ * @author Artem Vozhdayenko
+ * @author Matthias Thoma
+ *
+ * @since 6.0.1
+ *
+ */
+public class Mqttv5AdapterTests {
+
+	private void verifyUnsubscribe(IMqttAsyncClient client) throws Exception {
+		verify(client).connect(any(MqttConnectionOptions.class));
+		verify(client).subscribe(any(MqttSubscription[].class),any(), any(), any(IMqttMessageListener[].class), any());
+		verify(client).unsubscribe(any(String[].class));
+	}
+
+	private void verifyNotUnsubscribe(IMqttAsyncClient client) throws Exception {
+		verify(client).connect(any(MqttConnectionOptions.class));
+		verify(client).subscribe(any(MqttSubscription[].class),any(), any(), any(IMqttMessageListener[].class), any());
+		verify(client, never()).unsubscribe(any(String[].class));
+	}
+
+	private Mqttv5PahoMessageDrivenChannelAdapter buildAdapterIn(final IMqttAsyncClient client, Boolean cleanStart) throws MqttException {
```

Comment:
Why that cannot be a primitive `boolean`?

Why I'm OK with this method (which has to be `static`, BTW, and in the end of test class), I don't see a reason in other two.

Reply:
There is no particular reason. I took that code from MqttAdapterTests (the v3 version) and didn't want to change it too much to keep those two similar.